### ISSUE TITLE
faiss is always returning the same inde number for different queries

### ISSUE THREAD

**Issue by tianguang2525** [facebookresearch/faiss#3101]
# Summary
code:
res = faiss.StandardGpuResources() 
index_img = faiss.IndexFlatL2(image_tmp_emb.shape[-1])
index_img = faiss.index_cpu_to_gpu(res, 0, index_img)
index_img.add(retrieval.image_features)
image_features = model.model.encode_image(image, need_adapter_forward=args.need_hiddenAdap).float()
            
_, I = index_img.search(image_features.detach(), args.topk)

# Platform

OS: ubuntu

Faiss version: 1.7.2

Installed from:  anaconda

Faiss compilation options: 

Running on:
- [ ] CPU
- [1 ] GPU

Interface: 
- [ ] C++
- [1 ] Python

# Reproduction instructions

**Comment by tianguang2525** [comment]
when batch_size=8,the indexes are right.It is wrong when batch_size=64,128

**Comment by mdouze** [comment]
Can't repro this.
